### ISSUE TITLE
Error for `float32` data.

### ISSUE THREAD

**Issue by bazkiaei** [AstroHuntsman/huntsman-mocks#72]
The function `scale_light_by_distance` is one of the first functions that the user have to use to create the mock image and the particle positions are input to this function. I was trying to check the code with the different type (`dtype`) of `numpy.array` as its input and faced the following error when the input was `float32` array. It raises in the function `convert_to_lumonsity_distance` where the function uses the command `cosmo.luminosity_distance(redshifts)` to convert reshifts to luminosity distances.
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-33-34e616d80bb9> in <module>
      1 scaled, mscales = mocks.scale_light_by_distance(desired_data,
      2                                                 new_d_mass,
----> 3                                                 viewing_axis='x')

~/huntsman-mocks/mocks/mocks.py in scale_light_by_distance(particle_positions, particle_values, physical_distance, target_galaxy_comoving_depth, viewing_axis, config_location)
    121     particle_positions = convert_to_lumonsity_distance(particle_positions,
    122                                                        viewing_axis,
--> 123                                                        cosmo)
    124     # Applying corrections to the mass weights of particles.
    125     particle_values =\

~/huntsman-mocks/mocks/mocks.py in convert_to_lumonsity_distance(particle_positions, viewing_axis, cosmo)
    720         Dgrid.value, zgrid)
    721     particle_positions[:, AxisNumber[viewing_axis]] \
--> 722         = cosmo.luminosity_distance(zvals)
    723     return particle_positions
    724 

/anaconda3/lib/python3.7/site-packages/astropy/units/quantity.py in __setitem__(self, i, value)
   1048         if not self.isscalar and 'info' in self.__dict__:
   1049             self.info.adjust_indices(i, value, len(self))
-> 1050         self.view(np.ndarray).__setitem__(i, self._to_own_unit(value))
   1051 
   1052     # __contains__ is OK

/anaconda3/lib/python3.7/site-packages/astropy/units/quantity.py in _to_own_unit(self, value, check_precision)
   1300                                             value_dtype_array,
   1301                                             np.isnan(value_dtype_array))):
-> 1302                     raise TypeError("cannot convert value type to array type "
   1303                                     "without precision loss")
   1304         return _value

TypeError: cannot convert value type to array type without precision loss
```

**Comment by AnthonyHorton** [comment]
Before you try to fix this the first thing to do is work out where the 32 bit floats are coming from, and why. Is the simulation data 32 bit? Seems unlikely.

**Comment by bazkiaei** [comment]
> Before you try to fix this the first thing to do is work out where the 32 bit floats are coming from, and why. Is the simulation data 32 bit? Seems unlikely.

There are some codes that you can use to read the output of the simulations based on the GadGet code. I have not decided which one I want to use, but I am trying three of them and the output of all the three of them are `float32`.

**Comment by AnthonyHorton** [comment]
> There are some codes that you can use to read the output of the simulations based on the GadGet code. I have not decided which one I want to use, but I am trying three of them and the output of all the three of them are `float32`.

OK, so that's the root cause of the issue. Checking the user guide for GADGET-2 confirms that the standard output file format does indeed use 32 bit floats for positions (see page 32 of https://wwwmpa.mpa-garching.mpg.de/gadget/users-guide.pdf), so the Python libraries you are using to read those files are doing the right thing.

So, I think the best solution is simply to do an explicit conversion to 64 bit floats (using `.astype(np.float64)`) immediately after loading the data. That should stop the error, and ensure all subsequent calculations are done at double precision.

**Comment by bazkiaei** [comment]
Solved due to #73
